mcs/class/System/Mono.UnityTls/UnityTls.cs

```diff
@@ -0,0 +1,257 @@
+            public delegate void                                          unitytls_tlsctx_set_trace_callback_t(unitytls_tlsctx* ctx, unitytls_tlsctx_trace_callback cb, void* userData, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_set_trace_callback_t                   unitytls_tlsctx_set_trace_callback;
+            public delegate void                                          unitytls_tlsctx_set_x509verify_callback_t(unitytls_tlsctx* ctx, unitytls_tlsctx_x509verify_callback cb, void* userData, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_set_x509verify_callback_t              unitytls_tlsctx_set_x509verify_callback;
+            public delegate void                                          unitytls_tlsctx_set_supported_ciphersuites_t(unitytls_tlsctx* ctx, unitytls_ciphersuite* supportedCiphersuites, size_t supportedCiphersuitesLen, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_set_supported_ciphersuites_t           unitytls_tlsctx_set_supported_ciphersuites;
+            public delegate unitytls_ciphersuite                          unitytls_tlsctx_get_ciphersuite_t(unitytls_tlsctx* ctx, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_get_ciphersuite_t                      unitytls_tlsctx_get_ciphersuite;
+            public delegate unitytls_protocol                             unitytls_tlsctx_get_protocol_t(unitytls_tlsctx* ctx, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_get_protocol_t                         unitytls_tlsctx_get_protocol;
+            public delegate unitytls_x509verify_result                    unitytls_tlsctx_process_handshake_t(unitytls_tlsctx* ctx, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_process_handshake_t                    unitytls_tlsctx_process_handshake;
+            public delegate size_t                                        unitytls_tlsctx_read_t(unitytls_tlsctx* ctx, UInt8* buffer, size_t bufferLen, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_read_t                                 unitytls_tlsctx_read;
+            public delegate size_t                                        unitytls_tlsctx_write_t(unitytls_tlsctx* ctx, UInt8* data, size_t bufferLen, unitytls_errorstate* errorState);
+            public unitytls_tlsctx_write_t                                unitytls_tlsctx_write;
+            public delegate void                                          unitytls_tlsctx_free_t(unitytls_tlsctx* ctx);
+            public unitytls_tlsctx_free_t                                 unitytls_tlsctx_free;
+        }
+
+        [DllImport("__Internal")]
+        private static extern IntPtr mono_unity_get_unitytls_interface();
+        
+        private static mono_unity_unitytls_interface marshalledInterface = null;
+
+        public static bool IsSupported()
+        {
+            try {
+                return NativeInterface != null;
```

Comment:
It seems wasteful to try/catch in here, although I guess this is only called once. Any reason this can't be a little cleaner? Perhaps `NativeInterface` can check result of `mono_unity_get_unitytls_interface()` and return `null` if that is `null` rather than always doing the `Marshal.PtrToStructure` call and raising exception?

Reply:
My idea here was that I would also catch any exception thrown from `Marshal.PtrToStructure`. Looking at its possible exceptions again now I realize however that this should never happen so I agree with your suggestion. Will change accordingly!

Reply:
(done)